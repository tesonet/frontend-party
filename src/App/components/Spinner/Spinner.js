import React from 'react';
import styled, { keyframes } from 'styled-components';
import PropTypes from 'prop-types';

const Spinner = ({ size }) =>  (
  <SpinnerContainer>
    <SpinnerComponent size={size} />
  </SpinnerContainer>
);

const SpinnerContainer = styled.div`
  margin-top: 20px;
  display: flex;
  justify-content: center;
`;

const SpinAnimation = keyframes`
  0% { transform: rotate(0deg); }
  100% { transform: rotate(360deg); }
`

const SpinnerComponent = styled.div`
  display: inline-block;

  &::after {
    content: " ";
    display: block;
    width: ${props => props.size === 'large' ? '46px' : '30px'};
    height: ${props => props.size === 'large' ? '46px' : '30px'};
    margin: 1px;
    border-radius: 50%;
    border: 5px solid #cef;
    border-color: #cef transparent #cef transparent;
    animation: lds-dual-ring;
  }

  animation: ${SpinAnimation} 1.2s linear infinite;
`;

Spinner.propTypes = {
  size: PropTypes.string.isRequired,
}

export default Spinner;