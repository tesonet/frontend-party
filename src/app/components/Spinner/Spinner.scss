$color_bubbles_background: white;

@mixin bubbleAnimation($time) {
  -webkit-animation: pulse $time infinite;
  -moz-animation: pulse $time infinite;
  -ms-animation: pulse $time infinite;
  -o-animation: pulse $time infinite;
  animation: pulse $time infinite;
}

@mixin bubbleAnimationDelay($time) {
  -webkit-animation-delay: $time;
  -moz-animation-delay: $time;
  -ms-animation-delay: $time;
  -o-animation-delay: $time;
  animation-delay: $time;
}

.bubbles_container {
  margin: auto;
  width: 250px;
  height: 100px;
}

.bubbles {
  display: inline-block;
  height: 15px;
  width: 15px;
  margin: 18px 6px;

  border-radius: 100%;
  background-color: $color_bubbles_background;
  opacity: 0;
}

.bubble_1 {
  @include bubbleAnimation(0.9s);
}

.bubble_2 {
  @include bubbleAnimation(0.9s);
  @include bubbleAnimationDelay(0.15s);
}

.bubble_3 {
  @include bubbleAnimation(0.9s);
  @include bubbleAnimationDelay(0.3s);
}
