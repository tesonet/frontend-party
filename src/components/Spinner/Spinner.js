import styled, { keyframes } from 'styled-components';

const animation = keyframes`
    0% {
		transform: scale(0);
		opacity: 0;
	}
	50% {
		opacity: 1;
	}
	100% {
		transform: scale(1);
		opacity: 0;
	}
`;

export default styled.div`
  vertical-align: middle;
  margin: auto;
  border: 0.2em solid currentcolor;
  border-radius: 50%;
  animation: ${animation} 1s ease-out infinite;
  width: ${props => props.size};
  height: ${props => props.size};
  margin-top: ${props => (props.MTop ? props.MTop : 0)};
`;
