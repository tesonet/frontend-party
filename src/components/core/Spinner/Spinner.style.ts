import styled from 'styled-components'

export const SpinnerContent = styled.div`
  display: inline-block;
  width: 40px;
  height: 40px;

  &::after {
    content: ' ';
    display: block;
    width: 40px;
    height: 40px;
    margin: 8px;
    border-radius: 50%;
    border: 6px solid ${({ theme }) => theme.colors.limeade};
    border-color: ${({ theme }) => theme.colors.limeade} transparent
      ${({ theme }) => theme.colors.limeade} transparent;
    animation: lds-dual-ring 1.2s linear infinite;
  }
  @keyframes lds-dual-ring {
    0% {
      transform: rotate(0deg);
    }
    100% {
      transform: rotate(360deg);
    }
  }
`
