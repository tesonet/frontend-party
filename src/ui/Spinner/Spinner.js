import React from 'react'
import PropTypes from 'prop-types'
import styled, {keyframes} from 'styled-components'
import {space, variant} from 'styled-system'
import theme from '../styles/theme'

const sizeVariant = variant({
  prop: 'size',
  variants: {
    small: {
      width: 15,
      height: 15,
    },
    medium: {
      width: 20,
      height: 20,
    },
    large: {
      width: 40,
      height: 40,
    },
  },
})

const SVG_SIZE = 44

const keyframesRotate = keyframes`
  100% {
    transform: rotate(360deg);
  }
`

const keyframesStroke = keyframes`
  0% {
    stroke-dasharray: 1px, 200px;
    stroke-dashoffset: 0px;
  }
  50% {
    stroke-dasharray: 100px, 200px;
    stroke-dashoffset: -15px;
  }
  100% {
    stroke-dasharray: 100px, 200px;
    stroke-dashoffset: -125px;
  }
`

const SpinnerWrapper = styled.div`
  width: 20px;
  height: 20px;
  animation: ${keyframesRotate} 1.4s ease-in-out infinite;
  color: ${props => props.fillColor};
  ${space}
  ${sizeVariant}
`

const Circle = styled.circle`
  animation: ${keyframesStroke} 1.4s ease-in-out infinite;
  stroke-dasharray: 80px, 200px;
  stroke-dashoffset: 0px;
  stroke: currentColor;
`

const Spinner = ({color, size, thickness, ...other}) => {
  return (
    <SpinnerWrapper fillColor={color} size={size} role="progressbar" {...other}>
      <svg viewBox={`${SVG_SIZE / 2} ${SVG_SIZE / 2} ${SVG_SIZE} ${SVG_SIZE}`}>
        <Circle
          cx={SVG_SIZE}
          cy={SVG_SIZE}
          r={(SVG_SIZE - thickness) / 2}
          fill="none"
          strokeWidth={thickness}
        />
      </svg>
    </SpinnerWrapper>
  )
}

Spinner.propTypes = {
  color: PropTypes.string,
  size: PropTypes.oneOf(['small', 'medium', 'large']),
  thickness: PropTypes.number,
}

Spinner.defaultProps = {
  color: theme.palette.primary.main,
  size: 'medium',
  thickness: 4,
}

export default Spinner
