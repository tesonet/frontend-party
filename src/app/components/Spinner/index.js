import React from 'react';
import { serverSpinner } from '../../../assets';
import './Spinner.scss';

const Spinner = props => (
  <div>
    {props.spinnerType === 'loginSpinner' ? (
      <div className="bubbles_container">
        <div className="bubbles bubble_1" />
        <div className="bubbles bubble_2" />
        <div className="bubbles bubble_3" />
      </div>
    ) : props.spinnerType === 'serverSpinner' ? (
      <img src={serverSpinner} alt="Server Spinner" style={{ width: 100 }} />
    ) : (
      'Spinner type not defined.'
    )}
  </div>
);

export default Spinner;
