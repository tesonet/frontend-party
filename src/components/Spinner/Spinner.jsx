import React from "react";
import "./Spinner.scss";

const Spinner = () => <div className="loading" />;

export default Spinner;
