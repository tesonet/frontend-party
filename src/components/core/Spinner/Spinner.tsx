import React from 'react'
import { SpinnerContent } from './Spinner.style'

const Spinner: React.FC = () => {
  return <SpinnerContent />
}

export default Spinner
