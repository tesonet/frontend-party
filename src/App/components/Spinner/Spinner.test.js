import React from 'react';
import renderer from 'react-test-renderer';
import Spinner from './Spinner';

it('renders correctly small Spinner', () => {
  const SpinnerComponent = renderer
    .create(<Spinner size="small" />)
    .toJSON();
  expect(SpinnerComponent).toMatchSnapshot();
});

it('renders correctly large Spinner', () => {
  const SpinnerComponent = renderer
    .create(<Spinner size="large" />)
    .toJSON();
  expect(SpinnerComponent).toMatchSnapshot();
});