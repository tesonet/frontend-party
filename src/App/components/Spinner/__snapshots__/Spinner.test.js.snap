// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly large Spinner 1`] = `
<div
  className="sc-bdVaJa idoegx"
>
  <div
    className="sc-bwzfXH gxXqBo"
    size="large"
  />
</div>
`;

exports[`renders correctly small Spinner 1`] = `
<div
  className="sc-bdVaJa idoegx"
>
  <div
    className="sc-bwzfXH iWeIj"
    size="small"
  />
</div>
`;
